### PR TITLE
Fix setTags type on article setter

### PR DESCRIPTION
**Pull request by qsomazzi**
## Changelog

```markdown
### Changed
 - Update Article tags setter param type to be array 
```

## Subject

The goal of this PR is to fix the article tags setter to be an array. This allow fixtures to be implemented.

### CODE REVIEW

File: Model/AbstractArticle.php [hunk 1]
```diff
@@ -353,7 +353,7 @@ public function getSubtitle()
     /**
      * {@inheritdoc}
      */
-    public function setTags(Collection $tags = null)
+    public function setTags(array $tags = null)
```
Comment (greg0ire) [inline Model/AbstractArticle.php:356]: I think you can remove the `use` statement.

File: Model/ArticleInterface.php [hunk 2]
```diff
@@ -156,11 +156,11 @@ public function setSubtitle($subtitle);
...
-     * @param Tag[]|Collection $tags
+     * @param array $tags
      *
      * @return $this
      */
-    public function setTags(Collection $tags = null);
+    public function setTags(array $tags = null);
 
     /**
      * @return Tag[]|Collection $tags
```
Comment (greg0ire) [inline Model/ArticleInterface.php]: This should be changed, right?

**Review by greg0ire — CHANGES_REQUESTED** [review]

File: Model/AbstractArticle.php (hunk 1 above)
Comment (qsomazzi) [reply Model/AbstractArticle.php:356]: done

File: Model/AbstractArticle.php [hunk 3]
```diff
@@ -181,7 +180,7 @@ public function getAbstract()
     /**
      * {@inheritdoc}
      */
-    public function setCategories(Collection $categories = null)
+    public function setCategories(ArrayCollection $categories = null)
```
Comment (greg0ire) [inline Model/AbstractArticle.php]: I didn't notice that one. I don't think you should change to `ArrayCollection`, b/c it's better to type hint against an interface.

**Comment by qsomazzi** [comment]
@greg0ire all done :)

File: Model/AbstractArticle.php [hunk 4]
```diff
@@ -84,7 +84,7 @@
     protected $categories;
 
     /**
-     * @var Collection
+     * @var ArrayCollection
```
Comment (greg0ire) [inline Model/AbstractArticle.php:87]: Did you mean `array`?

Comment (qsomazzi) [reply Model/AbstractArticle.php:87]: no, It's still an ArrayCollection, just the setter change for the fixtures

Comment (greg0ire) [reply Model/AbstractArticle.php:87]: oh ok. Doctrine has its own magic to hydrate that IIRC, indeed

**Review by greg0ire — APPROVED** [review]

**Comment by OskarStark** [comment]
I mark this PR as `minor` instead of `major`, as there is no current release

**Pull request merged**